src/nomadic/summarize/main.py

```diff
@@ -0,0 +1,805 @@
+            n_field_passing=pd.NamedAgg("passing", lambda x: x.sum()),
+            per_field_contam=pd.NamedAgg("fail_contam", lambda x: 100 * x.mean()),
+            per_field_lowcov=pd.NamedAgg("fail_lowcov", lambda x: 100 * x.mean()),
+            per_field_passing=pd.NamedAgg("passing", lambda x: 100 * x.mean()),
+        )
+        .reset_index()
+    )
+    final_df.to_csv(f"{output_dir}/summary.experiments_qc.csv", index=False)
+
+    # --------------------------------------------------------------------------------
+    # Let's move onto to variant calling results
+    #
+    # --------------------------------------------------------------------------------
+
+    log.info("Loading variants...")
+    variant_df = load_and_concat_variants(expt_dirs)
+
+    if "sample_id" in variant_df.columns:
+        variant_df.drop(columns=["sample_id"], inplace=True)
+    # Merge with the quality control results, then we can subset to the analysis set
+    variant_df = pd.merge(
+        left=coverage_df.rename({"name": "amplicon"}, axis=1)[
+            ["expt_name", "barcode", "sample_id", "sample_type", "amplicon", "status"]
+        ],
+        right=variant_df,
+        on=["expt_name", "barcode", "amplicon"],
+    )
+
+    log.info("Filtering to analysis set...")
+    remove_genes = ["hrp2", "hrp3"]  # noqa: F841 later used in query
+    remove_mutations = ["crt-N75K"]  # noqa: F841 later used in query
+    analysis_df = (
+        variant_df.query("status == 'pass'")
+        .query("mut_type == 'missense'")
+        .query("gene not in @remove_genes")
+        .query("mutation not in @remove_mutations")
+    )
+
+    # Filter out false positives
+    analysis_df = filter_false_positives(analysis_df)
+    analysis_df.to_csv(f"{output_dir}/summary.variants.analysis_set.csv", index=False)
+
+    # Then we will compute prevalence
+    prev_df = compute_variant_prevalence(analysis_df)
+    prev_df.to_csv(f"{output_dir}/summary.variants.prevalence.csv", index=False)
+
+    for col in prevalence_by:
+        prev_by_col_df = compute_variant_prevalence_per(
+            analysis_df, master_metadata, [col]
+        )
+        prev_by_col_df.to_csv(
+            f"{output_dir}/summary.variants.prevalence-{col}.csv", index=False
+        )
+
+    # --------------------------------------------------------------------------------
+    # Gene deletion analysis
+    #
+    # --------------------------------------------------------------------------------
+
+    log.info("Calculate gene deletions...")
+    gene_deletion_df = gene_deletions(coverage_df, ["hrp2", "hrp3"])
+    gene_deletion_df.to_csv(f"{output_dir}/summary.gene_deletions.csv", index=False)
+
+    prev_gen_deletions_df = gene_deletion_prevalence_by(
+        gene_deletion_df, master_metadata, []
+    )
+    prev_gen_deletions_df.to_csv(
+        f"{output_dir}/summary.gene-deletions.prevalence.csv", index=False
+    )
+
+    for col in prevalence_by:
```

Comment:
I think we should check that the `prevalence_by` columns are all in the metadata further up, and provide some graceful error handling if they are not.